datafusion/catalog/src/table.rs

```diff
@@ -273,6 +273,7 @@ pub trait TableProvider: Sync + Send {
         _state: &dyn Session,
         _input: Arc<dyn ExecutionPlan>,
         _overwrite: bool,
+        _replace_into: bool,
```

Comment:
Can you please document this new parameter and explain what it means (bonus points for documenting what `_overwrite` means too)

Reply:
> That shows that a mock up user defined TableProvider will have insert_into called with the correct value of replace_into when the relevant SQL is planned

Test added in `datafusion/core/tests/user_defined/insert_operation.rs`.

> Can you please document this new parameter and explain what it means (bonus points for documenting what _overwrite means too)

Documentation added to the variants of the `InsertOp` enum.